x/marketplace/client/cli/tx_mint_nft.go

```diff
@@ -0,0 +1,70 @@
+package cli
+
+import (
+	"strconv"
+
+	"github.com/CudoVentures/cudos-node/x/marketplace/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+)
+
+var _ = strconv.Itoa(0)
+
+func CmdMintNft() *cobra.Command {
```

Comment:
> I am getting a bit confused between nft and collection objects in both x/marketplace and x/nft module. Can't we reuse them and not have two? Also apart from that the general structure ( as noted in the other PR - #45 ) is not as the standard cosmos-sdk module and our other modules. I have some other questions that we can discuss.

Object has different meaning in different business context. I don't see what value we would get by trying to re-use structs from different domain. Even if we ignore that, it will be messy because the ones from NFT module have properties we don't need in marketplace module.

Reply:
> Why are we minting from here instead of the nft module ?

This is for the special case where the clients explicitly requested on-demand minting, here the minting will be executed only by whitelisted address specified in the minter property of the denom. There is no other reasonable way to accomplish the on-demand minting.